api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -67,17 +77,44 @@ impl<S: ApiServerStorage + Send + Sync> LocalBlockchainState for BlockchainState
             db_tx.del_main_chain_block_id(current_best.0).await?;
         }
 
+        // Disconnect address balances
+        db_tx.del_address_balance_above_height(common_block_height).await?;
+
         // Connect the new blocks in the new chain
         for (index, block) in blocks.into_iter().map(WithId::new).enumerate() {
             let block_height = BlockHeight::new(common_block_height.into_int() + index as u64 + 1);
 
             db_tx.set_main_chain_block_id(block_height, block.get_id()).await?;
             logging::log::info!("Connected block: ({}, {})", block_height, block.get_id());
 
+            update_balances_from_outputs(
+                Arc::clone(&self.chain_config),
+                &mut db_tx,
+                block_height,
+                block.block_reward().outputs(),
+            )
+            .await?;
```

Comment:
There are also inputs in the block header.

Reply:
Oh, ok thanks for the heads up.

Reply:
I've just checked, and I can't find it. Did you mean the block reward here? If so, it's being taken care of with:

```
           update_balances_from_outputs(
                Arc::clone(&self.chain_config),
                &mut db_tx,
                block_height,
                block.**block_reward()**.outputs(),
            )
            .await
            .expect("Unable to update balances from block reward outputs");
```

Reply:
in ConsensusData, for proof of stake, there are inputs.

Reply:
I've just had a look, and I think PoS handling should go in the next PR for pools... the reward for PoS is `CreateStakePool|ProduceBlockFromStake` and both don't contain the output value. I haven't gotten there yet in the pool code, but I'm guessing the PoS reward is calculated on decommissioning.

Reply:
Alrighty.